In [46]:
import math
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from scipy.stats import bernoulli

In [2]:
df = pd.read_csv('./data/train.csv')

In [3]:
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

### Columns to generate:
- Survived
- Age
- SibSp
- Parch
- Fare
- Sex
- Pclass
- Embarked

### Generating Pclass:

In [43]:
df['Pclass'].value_counts() / len(df)

3    0.551066
1    0.242424
2    0.206510
Name: Pclass, dtype: float64

We see that about 55.1% of passengers were class 3, 24.24% of passengers were class 1, and 20.65% were class 2.

### Data Generating Process for Assigning Pclass:
- Let $X\sim N(0.24, 0.05)$ and $Y\sim N(0.206, 0.05)$
- $\left \lfloor{891*X}\right \rfloor$ will be the number of passengers in Pclass1 and $\left \lfloor{891*Y}\right \rfloor$ be the number of passengers in Pclass2
- $891 - \left \lfloor{891*X}\right \rfloor - \left \lfloor{891*Y}\right \rfloor$ will be the remaining in Pclass3.

# Joint distributions?

In [59]:
num_pclass1 = math.floor(np.random.normal(0.24, 0.05) * 891)
num_pclass2 = math.floor(np.random.normal(0.206, 0.05) * 891)
num_pclass3 = 891 - num_pclass1 - num_pclass2

In [ ]:
np.zeros(20, dtype=int)
K[:10] = 1
np.random.shuffle(K)
print(K)

In [67]:
x = np.zeros(5)

In [88]:
df.loc[df['Pclass'] == 1]['Sex'].value_counts()

male      122
female     94
Name: Sex, dtype: int64

In [89]:
df.loc[df['Pclass'] == 2]['Sex'].value_counts()

male      108
female     76
Name: Sex, dtype: int64

In [90]:
df.loc[df['Pclass'] == 3]['Sex'].value_counts()

male      347
female    144
Name: Sex, dtype: int64

In [28]:
number_passengers = len(df)

In [29]:
df.groupby('Pclass').agg({'Survived': lambda x: sum(x) / len(x)})

Survived
Pclass          
1       0.629630
2       0.472826
3       0.242363

In [34]:
df.groupby('Sex').agg({'Survived': lambda x: sum(x) / len(x)})

Survived
Sex             
female  0.742038
male    0.188908

In [ ]:
plt.plot(df)

In [26]:
bernoulli.rvs(size=5, p=0.8)

array([0, 1, 1, 0, 1])

In [4]:
df['Pclass']

0      3
1      1
2      3
3      1
4      3
      ..
886    2
887    1
888    3
889    1
890    3
Name: Pclass, Length: 891, dtype: int64

---

In [97]:
!pip install sdv

     |████████████████████████████████| 102 kB 1.1 MB/s ta 0:00:01
     |████████████████████████████████| 131 kB 5.4 MB/s eta 0:00:01
     |████████████████████████████████| 61 kB 450 kB/s  eta 0:00:01
     |████████████████████████████████| 47 kB 5.1 MB/s  eta 0:00:01
     |████████████████████████████████| 53 kB 1.9 MB/s  eta 0:00:01
     |████████████████████████████████| 1.6 MB 32.4 MB/s eta 0:00:01
     |████████████████████████████████| 890.2 MB 11 kB/s s eta 0:00:012     |████████████                    | 334.7 MB 45.8 MB/s eta 0:00:13     |█████████████▎                  | 368.8 MB 39.3 MB/s eta 0:00:14     |███████████████▉                | 440.7 MB 36.8 MB/s eta 0:00:13     |████████████████▍               | 456.2 MB 36.8 MB/s eta 0:00:12     |█████████████████████▋          | 601.8 MB 140.0 MB/s eta 0:00:03     |████████████████████████████    | 777.2 MB 37.7 MB/s eta 0:00:03     |█████████████████████████████▊  | 827.3 MB 3.7 MB/s eta 0:00:17
     |████████████████████████

In [98]:
from sdv import SDV

ModuleNotFoundError: No module named 'sdv'

In [ ]:
sdv = SDV()

sdv.fit(metadata, tables)